### ISSUE TITLE
Get resulting file as File or Bin data for additional using

### ISSUE THREAD

**Issue by AstRonin** [eKoopmans/html2pdf#12]
This function created PDF and save to a file
```javascript
html2pdf.makePDF(canvas, pageSize, opt);
```
How can I to get PDF as File, for example, if I want to send it to email?

**Comment by eKoopmans** [comment]
Good question! Right now, you can't. It would be a useful feature, and I'll add it to the list of improvements to make when I revamp the API.

**Comment by eKoopmans** [comment]
Hi, I have just pushed a new branch, [new-api](https://github.com/eKoopmans/html2pdf/tree/new-api), which is ready for testing! [The readme](https://github.com/eKoopmans/html2pdf/tree/new-api#advanced-usage) has all the information on the new API. Please let me know if there are any problems with it - I want to make sure it's stable before merging into master!

For this specific request, you could use the `.output` method to get a datauristring:
```js
html2pdf().from(element).output('datauristring').then(function (val) {
  // Your code to send an email with val.
});
```

**Comment by khateeburrehman** [comment]
> Hi, I have just pushed a new branch, [new-api](https://github.com/eKoopmans/html2pdf/tree/new-api), which is ready for testing! [The readme](https://github.com/eKoopmans/html2pdf/tree/new-api#advanced-usage) has all the information on the new API. Please let me know if there are any problems with it - I want to make sure it's stable before merging into master!
> 
> For this specific request, you could use the `.output` method to get a datauristring:
> 
> ```js
> html2pdf().from(element).output('datauristring').then(function (val) {
>   // Your code to send an email with val.
> });
> ```

Hi, the Above method downloads file too, i only need uristring (val). How to set it 
**download:false**

**Comment by nauman555** [comment]
> This function created PDF and save to a file
> 
> ```js
> html2pdf.makePDF(canvas, pageSize, opt);
> ```
> 
> How can I to get PDF as File, for example, if I want to send it to email?

@AstRonin  did you find any solution for this ?

**Comment by AstRonin** [comment]
> > This function created PDF and save to a file
> > ```js
> > html2pdf.makePDF(canvas, pageSize, opt);
> > ```
> > 
> > 
> > How can I to get PDF as File, for example, if I want to send it to email?
> 
> @AstRonin did you find any solution for this ?

Autor wrote changes, please look. https://github.com/eKoopmans/html2pdf/issues/12#issuecomment-368236593
Sorry, but I cannot test it right now...

**Comment by eKoopmans** [comment]
Hi everyone, my response [last February](https://github.com/eKoopmans/html2pdf/issues/12#issuecomment-368236593) is still valid, and should give you the file as a "data URI" string. Note it's the same solution I've posted [here](https://github.com/eKoopmans/html2pdf/issues/54#issuecomment-453027494) and other places. @khateeburrehman that code should not prompt the user to download the file, there must be something else in your code causing that.

Closing this issue as resolved by the v0.9.0 new API.
